lib/crystalball/graphql/map_generator/graphql_strategy/patch.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Crystalball
+  module GraphQL
+    module MapGenerator
+      class GraphQLStrategy
+        # Patches `GraphQL::ObjectType` to track object definitions
+        module Patch
+          class << self
+            # Patches `GraphQL::ObjectType.define`. Renames original `define` to `old_define` and
+            # replaces it with custom one
+            def apply!
+              ::GraphQL::ObjectType.class_eval do
+                class << self
+                  include Patch
+
+                  alias_method :cb_original_define, :define
+                  alias_method :define, :cb_patched_define
+                end
+              end
+            end
+
+            # Reverts original behavior of `GraphQL::ObjectType#define`
+            def revert!
+              ::GraphQL::ObjectType.class_eval do
+                class << self
+                  alias_method :define, :cb_original_define # rubocop:disable Lint/DuplicateMethods
+                  undef_method :cb_patched_define
+                end
+              end
+            end
+          end
+
+          # Will replace original `GraphQL::ObjectType#define`.
+          # Pushes path of files where `ObjectType`s are defined.
+          # and then calls the original `define`
+          def cb_patched_define(*args, **kwargs, &block)
+            GraphQLStrategy.type_definition_paths.concat(caller_locations.map(&:path))
```

Comment:
It will return huge list. Why do you need it?

Reply:
Indeed. It's being filtered in `GraphQLStrategy#call` but we still keep it in the class instance variable. Do you know a better way to get the definition paths? Or maybe it's better to just filter it here